src/traces/bar/hover.js

```diff
@@ -161,7 +160,7 @@ function hoverOnBars(pointData, xval, yval, hovermode) {
     pointData.valueLabel = hoverLabelText(sa, pointData[sizeLetter + 'LabelVal']);
 
     // spikelines always want "closest" distance regardless of hovermode
-    pointData.spikeDistance = (sizeFn(di) + thisBarPositionFn(di)) / 2 + maxSpikeDistance - maxHoverDistance;
+    pointData.spikeDistance = (sizeFn(di) + thisBarPositionFn(di)) / 2 - maxHoverDistance;
```

Comment:
`maxHoverDistance` is necessary here because it is added in the function `sizeFn`. Now, why would a function compute the size of a bar would contain `maxHoverDistance` is beyond my comprehension.

Reply:
Yeah, super confusing. It's not computing the size of a bar, it's computing a distance measure along the size axis. Is it necessary to remove `maxSpikeDistance` here? Or it just doesn't have an understood purpose?

Reply:
> Is it necessary to remove maxSpikeDistance here?

Keeping `maxSpikeDistance` in leads to:
![Screenshot_2020-03-13_15-47-32](https://user-images.githubusercontent.com/301546/76654748-1acb4f00-6542-11ea-9a4b-e0ff1d2851f4.png)

> Or it just doesn't have an understood purpose?

I'll be honest: I also don't understand its purpose. What is `spikedistance` supposed to represent? Why would we add `maxSpikeDistance` and substract `maxHoverDistance` to it at the trace-level? It seems to me that this logic should live in `fx/hover.js` no?

Reply:
yeah this logic should certainly be cleaned up at some point... super hacky. Perhaps as part of #4638 🙏